pkg/local_object_storage/blobstor/fstree/fstree.go

```diff
@@ -354,6 +356,44 @@ func (t *FSTree) Put(addr oid.Address, data []byte) error {
 	return nil
 }
 
+// PutBatch puts a batch of objects in the storage.
+func (t *FSTree) PutBatch(objs []*objectSDK.Object) error {
+	if t.readOnly {
+		return common.ErrReadOnly
+	}
+
+	putPrms := make([]putPrm, 0, len(objs))
+	for _, obj := range objs {
+		p := t.treePath(object.AddressOf(obj))
+		if err := util.MkdirAllX(filepath.Dir(p), t.Permissions); err != nil {
+			return fmt.Errorf("mkdirall for %q: %w", p, err)
+		}
+		putPrms = append(putPrms, putPrm{
+			id:   obj.GetID(),
+			path: p,
+			data: t.Compress(obj.Marshal()),
+		})
+	}
+
+	_, ok := t.writer.(*genericWriter)
```

Comment:
It can be handled in `func (w *genericWriter) writeBatch(objs []putPrm) error`, you don't need this is the common code.